graphula-core/src/Graphula/Internal.hs

```diff
@@ -51,6 +51,11 @@ class GHasDependenciesRecursive fieldsProxy node deps where
   genericDependsOnRecursive :: fieldsProxy -> node -> deps -> node
 
 -- This instance head only matches EoT representations of
+-- datatypes with no constructors. Ignore dependencies.
+instance GHasDependencies (Proxy nodeTy) (Proxy depsTy) Void deps where
```

Comment:
Hmmm, do we need to cover all cases? Or just the cases that we support?

---

This also raises the question of `Void`. I considered it when thinking about the default `Dependencies`. Here are my thoughts:

`Void` is idiomatically used when you are trying to explicitly outlaw a codepath. For example in `pipes` we use `Void` to express that something will never `yield`. So what does it mean to have `type Dependencies a = Void`?

```
x `dependsOn` y = const x $ absurd y
```

This could really be reduced to just `dependsOn = const`. The same instance for `()`.

In order to write `node` we'd need to introduce a bottom.

```
node = nodeWith $ error "can never have dependencies"
```

So, it might be "morally" correct to use `Void`. in a sense it is absurd for something with no dependencies to depend on anything. We should never being inspecting non-existent dependencies.

`()` on the other hand plays nicely into the algebra of types. `()` is the identity object of product types. It erases itself by carrying no semantic meaning. However, there really is nothing in graphula forcing us to produce just product types and then `()` is inappropriate because it does carry meaning in association with sum types.

---

`type Dependencies a = Void` also has a nice property that it forces `nodeWith` to be impossible to utilize without a bottom, so users wouldn't accidentally litter `nodeWith ()` around their codebase.